src/dialogs/more-info/controls/more-info-media_player.ts

```diff
@@ -0,0 +1,381 @@
+      return supportsFeature(stateObj, SUPPORTS_PLAY)
+        ? [
+            {
+              icon: "hass:play",
+              action: "media_play",
+            },
+          ]
+        : undefined;
+    }
+
+    const buttons: ControlButton[] = [];
+
+    if (supportsFeature(stateObj, SUPPORT_TURN_OFF)) {
+      buttons.push({
+        icon: "hass:power",
+        action: "turn_off",
+      });
+    }
+
+    if (supportsFeature(stateObj, SUPPORT_PREVIOUS_TRACK)) {
+      buttons.push({
+        icon: "hass:skip-previous",
+        action: "media_previous_track",
+      });
+    }
+
+    if (
+      (state === "playing" &&
+        (supportsFeature(stateObj, SUPPORT_PAUSE) ||
+          supportsFeature(stateObj, SUPPORT_STOP))) ||
+      (state === "paused" && supportsFeature(stateObj, SUPPORTS_PLAY))
+    ) {
+      buttons.push({
+        icon:
+          state !== "playing"
+            ? "hass:play"
+            : supportsFeature(stateObj, SUPPORT_PAUSE)
+            ? "hass:pause"
+            : "hass:stop",
+        action: "media_play_pause",
+      });
+    }
+
+    if (supportsFeature(stateObj, SUPPORT_NEXT_TRACK)) {
+      buttons.push({
+        icon: "hass:skip-next",
+        action: "media_next_track",
+      });
+    }
+
+    return buttons.length > 0 ? buttons : undefined;
+  }
+
+  private _handleClick(e: MouseEvent): void {
+    this.hass!.callService(
+      "media_player",
+      (e.currentTarget! as HTMLElement).getAttribute("action")!,
+      {
+        entity_id: this.stateObj!.entity_id,
+      }
+    );
+  }
+
+  private _toggleMute() {
+    this.hass!.callService("media_player", "volume_mute", {
+      entity_id: this.stateObj!.entity_id,
+      is_volume_muted: !this.stateObj!.attributes.is_volume_muted,
+    });
+  }
+
+  private _selectedValueChanged(e: Event): void {
+    this.hass!.callService("media_player", "volume_set", {
+      entity_id: this.stateObj!.entity_id,
+      volume_level:
+        Number((e.currentTarget! as HTMLElement).getAttribute("value")!) / 100,
+    });
+  }
+
+  private _handleSourceChanged(e: CustomEvent) {
+    const newVal = e.detail.value;
+
+    if (!newVal || this.stateObj!.attributes.source === newVal) return;
+
+    this.hass.callService("media_player", "select_source", {
+      source: newVal,
+    });
+  }
+
+  private _handleSoundModeChanged(e: CustomEvent) {
+    const newVal = e.detail.value;
+
+    if (!newVal || this.stateObj?.attributes.sound_mode === newVal) return;
+
+    this.hass.callService("media_player", "select_sound_mode", {
+      sound_mode: newVal,
+    });
+  }
+
+  private _ttsCheckForEnter(e: KeyboardEvent) {
+    if (e.keyCode === 13) this._sendTTS();
+  }
+
+  private _sendTTS() {
+    const ttsInput = this._ttsInput;
+    if (!ttsInput) {
+      return;
+    }
+
+    const services = this.hass.services.tts;
+    const serviceKeys = Object.keys(services).sort();
+
+    const service = serviceKeys.find((key) => key.indexOf("_say") !== -1);
```

Comment:
@bramkragten This should be a bit better than the for loop I think. Still maybe not the best way

Reply:
the fact that we just pick the first one is wrong, to begin with... But yeah, this code is a lot better

Reply:
Yea... that's what I was thinking. We also sort them... but from the data, we are given from the service there doesn't seem to be a way to find the preferred one